internal/verifier/recheck.go

```
@@ -137,6 +130,12 @@ func (verifier *Verifier) insertRecheckDocs(
 						options.BulkWrite().SetOrdered(false),
 					)
 
+					err = util.TolerateSimpleDupeKeyInBulk(
```

Where does the first recheck document come from? It makes sense to ignore the error if it only comes from the last retry or a previous MV run. Could we add a comment here to explain?

---

I’ve added this comment:
```
+                                       // We expect duplicate-key errors from the above because:
+                                       //
+                                       // a) The same document can change multiple times per generation.
+                                       // b) The source’s changes also happen on the destination, and both
+                                       //    change streams populate this recheck queue.
+                                       //
+                                       // Because of that, we ignore duplicate-key errors. We do *not*, though,
+                                       // ignore other errors that may accompany duplicate-key errors in the
+                                       // same server response.
```

---

Thanks for the explanation